logstash-core/src/main/java/org/logstash/config/ir/compiler/AbstractOutputDelegatorExt.java

```diff
@@ -104,14 +104,17 @@ public IRubyObject multiReceive(final IRubyObject events) {
         return this;
     }
 
-    protected void initMetrics(final String id, final AbstractMetricExt metric) {
+    protected void initMetrics(final String id, final AbstractMetricExt metric, final String codeRef) {
```

Comment:
hmm, seems like you will always convert this arg to a `RubyString` why not pass down a `RubyString` ?

Reply:
No in case of `JavaOutpoutDelegatorExt.create` the string is created from Java coda